### PR TITLE
feat: custom scalar support

### PR DESCRIPTION
**Pull request by miwialex**
Currently there is no ability to add support for custom scalars for example `Map`. This gives the ability to add mocks for your custom scalars. an example setup would be in your options object to do the following.

```javascript
options = {
  scalarMocks: {
    Map: () => {
      return { foo: 'bar' }
    }
  }
}
```
Then anytime a custom scalar of Map is called, it resolves using the specified resolver.

### CODE REVIEW

**Comment by jgwhite** [comment]
This was much simpler than I expected!
I guess to round this out we’ll wanna add test cases and documentation in the README.

**Comment by jgwhite** [comment]
Addresses #19

File: addon/mocks/create.js [hunk 1]
```diff
@@ -18,6 +18,10 @@ export const composeCreateMocksForSchema =
 
       let mocks = createMocks(typesAndMockFns, db, options);
 
+      if(options.scalars) {
+        mocks = Object.assign({}, mocks, options.scalars)
+      }
```
Comment (jneurock) [inline addon/mocks/create.js]: 2 quick notes here:
1. Please check for options. They are allowed to be null.
2. Please format the code slightly so it matches the rest of the codebase.

```javascript
if (options && options.scalars) {
  mocks = Object.assign({}, mocks, options.scalars);
}
```

**Comment by jneurock** [comment]
Thanks for the PR!

I would like to suggest that the option name be a little more specific. I think `scalarMocks` would be great.

**Comment by miwialex** [comment]
> Thanks for the PR!
> 
> I would like to suggest that the option name be a little more specific. I think `scalarMocks` would be great.

Thanks for the feedback @jneurock ! I made the updates you requested and updated the README. I'm not sure why the build is failing though, as I didn't touch any of the dependencies.

**Comment by jneurock** [comment]
One particular Ember Try scenario has been failing for a while. I need to update some things. Not your fault.

How do you feel about adding an acceptance test? You could extend an existing test module or add a new one. Since I haven't tested custom scalars in my own apps, it would be cool to see this in action. The unit test doesn't quite tell me if the feature works.

**Comment by jneurock** [comment]
Thanks again for the PR!

**Comment by jgwhite** [comment]
🙌 y’all are the best 🙌

**Pull request merged**